src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/updatefrc/generators/UpdateFrcInfoLetterDetailsGenerator.java

```diff
@@ -0,0 +1,112 @@
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class UpdateFrcInfoLetterDetailsGenerator {
+
+    private final DocumentHelper documentHelper;
+    private final CaseDataService caseDataService;
+    public static final String LETTER_DATE_FORMAT = "yyyy-MM-dd";
+
+    public UpdateFrcInfoLetterDetails generate(CaseDetails caseDetails,
+                                               DocumentHelper.PaperNotificationRecipient recipient) {
+        return UpdateFrcInfoLetterDetails.builder()
+            .courtDetails(buildFrcCourtDetails(caseDetails.getData()))
+            .reference(getSolicitorReference(caseDetails, recipient))
+            .divorceCaseNumber(Objects.toString(caseDetails.getData().get(DIVORCE_CASE_NUMBER)))
+            .applicantName(documentHelper.getApplicantFullName(caseDetails))
+            .respondentName(documentHelper.getRespondentFullNameContested(caseDetails))
+            .addressee(getAddressee(caseDetails, recipient))
+            .caseNumber(caseDetails.getId().toString())
+            .letterDate(DateTimeFormatter.ofPattern(LETTER_DATE_FORMAT).format(LocalDate.now()))
+            .build();
+    }
+
+    private Addressee getAddressee(CaseDetails caseDetails, DocumentHelper.PaperNotificationRecipient recipient) {
+        Map<String, Object> caseData = caseDetails.getData();
+
+        if (isApplicantSolicitor(caseData, recipient)) {
+            log.info("Recipient is Applicant's Solicitor");
+            return buildAddressee(nullToEmpty((caseData.get(CONTESTED_SOLICITOR_NAME))),
+                getSolicitorFormattedAddress(caseDetails, CONTESTED_SOLICITOR_ADDRESS));
+        } else if (isRespondentSolicitor(caseData, recipient)) {
+            log.info("Recipient is Respondent's Solicitor");
+            return buildAddressee(nullToEmpty((caseData.get(RESP_SOLICITOR_NAME))),
+                getSolicitorFormattedAddress(caseDetails, RESP_SOLICITOR_ADDRESS));
+        }
+
+        log.info("Recipient is {}", recipient);
+        return buildAddressee(getLitigantName(caseDetails, recipient), getLitigantFormattedAddress(caseDetails, recipient));
+    }
+
+    private Addressee buildAddressee(String name, String address) {
+        return Addressee.builder()
+            .name(name)
+            .formattedAddress(address)
+            .build();
+    }
+
+    private String getLitigantName(CaseDetails caseDetails, DocumentHelper.PaperNotificationRecipient recipient) {
+        boolean isApplicant = recipient == APPLICANT;
+        return isApplicant
+            ? caseDataService.buildFullApplicantName(caseDetails)
+            : caseDataService.buildFullRespondentName(caseDetails);
+    }
+
+    private String getLitigantFormattedAddress(CaseDetails caseDetails,
+                                               DocumentHelper.PaperNotificationRecipient recipient) {
+        Map<String, Object> caseData = caseDetails.getData();
+        boolean isApplicant = recipient == APPLICANT;
```

Comment:
I think this isn't needed, a bit confusing = ==, I'd rather extract that ternary to a method